anvill/python/anvill/arch.py

```diff
@@ -13,32 +13,45 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+from abc import ABC, abstractmethod
 
-class Arch(object):
+
+class Arch(ABC):
     """Generic architecture description."""
 
-    def name(self):
-        raise NotImplementedError()
+    @abstractmethod
+    def name(self) -> str:
+        ...
 
-    def program_counter_name(self):
-        raise NotImplementedError()
+    @abstractmethod
+    def program_counter_name(self) -> str:
+        ...
 
-    def stack_pointer_name(self):
-        raise NotImplementedError()
+    @abstractmethod
+    def stack_pointer_name(self) -> str:
+        ...
 
-    def return_address_proto(self):
-        raise NotImplementedError()
+    @abstractmethod
+    def return_address_proto(self) -> dict:
+        ...
 
-    def return_stack_pointer_proto(self, num_bytes_popped):
-        raise NotImplementedError()
+    @abstractmethod
+    def return_stack_pointer_proto(self, num_bytes_popped) -> dict:
+        ...
 
-    def pointer_size(self):
-        raise NotImplementedError()
+    @abstractmethod
+    def pointer_size(self) -> int:
+        ...
 
-    def register_family(self, reg_name):
-        raise NotImplementedError()
+    @abstractmethod
+    def stack_offset(self) -> int:
+        ...
 
-    def register_name(self, reg_name):
+    @abstractmethod
+    def register_family(self, reg_name) -> tuple:
```

Comment:
It should be `Tuple` (also needs importing) and maybe something like: `Tuple[blah, ...]` where the `blah` is the type name for the first thing, and the `...` signals that there is a variable number of things in the tuple, and each subsequent thing has the same type as `blah`.